sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -717,47 +717,91 @@ gnrc_netif_t *gnrc_netif_get_by_prefix(const ipv6_addr_t *prefix)
     return best_netif;
 }
 
+static int _netif_ops_set_helper(gnrc_netif_t *netif, netopt_t opt,
+                                 void *data, uint16_t data_len)
+{
+    gnrc_netapi_opt_t netapi_opt = {
+        .opt = opt,
+        .data = data,
+        .data_len = data_len,
+    };
+    return netif->ops->set(netif, &netapi_opt);
+}
+
 int gnrc_netif_ipv6_group_join_internal(gnrc_netif_t *netif,
                                         const ipv6_addr_t *addr)
 {
-    unsigned idx = UINT_MAX;
+    uint8_t group_data[GNRC_NETIF_L2ADDR_MAXLEN];
+    int res;
 
+    /* can be called out of lock */
+    res = gnrc_netif_ipv6_group_to_l2_group(netif, addr, group_data);
     gnrc_netif_acquire(netif);
+    if (res > 0) {
+
+        res = _netif_ops_set_helper(netif, NETOPT_L2_GROUP,
+                                    group_data, (uint16_t)res);
+        /* link layer does not support multicast, but we can still use
+         * broadcast */
+        if ((res < 0) && (res != -ENOTSUP)) {
+            goto end;
+        }
+    }
+    /* link layer does not support multicast, but we can still use
+     * broadcast */
```

Comment:
move that to line 791, already one of those comments right above :wink:

Reply:
Please rephrase.. It is not really clear to me, what you mean.

Reply:
`gnrc_netif_ipv6_group_join_internal()` has two times the same comment, `gnrc_netif_ipv6_group_leave_internal()` has none - I'm suggesting a more equal distribution ;)

Reply:
Both comments in join are justified, but I'll add to leave... testing revealed another bug in leave that I am currently fixing. Will add it.